--- a/luci-app-fancontrol/Makefile
+++ b/luci-app-fancontrol/Makefile
@@ -7,10 +7,10 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI support for Fan Controller
-LUCI_DEPENDS:=+luci +fancontrol
+LUCI_DEPENDS:=+luci +fancontrol +kmod-hwmon-pwmfan
 LUCI_PKGARCH:=all
 
 PKG_NAME:=luci-app-fancontrol

--- a/luci-app-fan/Makefile
+++ b/luci-app-fan/Makefile
@@ -6,7 +6,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI fan control
-LUCI_DEPENDS:=+coreutils-stat
+LUCI_DEPENDS:=+coreutils-stat +kmod-hwmon-pwmfan
 LUCI_PKGARCH:=all
 PKG_MAINTAINER:=jjm2473 <jjm2473@gmail.com>
 
