--- a/luci-app-my-dnshelper/htdocs/luci-static/resources/view/my-dnshelper/filters.js
+++ b/luci-app-my-dnshelper/htdocs/luci-static/resources/view/my-dnshelper/filters.js
@@ -38,12 +38,12 @@ return view.extend({
 		s.anonymous = true;
 
 		o = s.option(form.DynamicList, "url", _("Filter Rules Subscription URL"));
-		o.value("https://cdn.jsdelivr.net/gh/privacy-protection-tools/anti-AD@master/adblock-for-dnsmasq.conf", _("anti-AD (Privacy-Protect|Preferred)"));
-		o.value("https://gcore.jsdelivr.net/gh/AdguardTeam/AdGuardSDNSFilter@gh-pages/Filters/filter.txt", _("AdGuard Filter"));
-		o.value("https://gcore.jsdelivr.net/gh/Cats-Team/AdRules/hosts.txt", _("Cats-Team Hosts"));
-		o.value("https://cdn.jsdelivr.net/gh/VeleSila/yhosts/hosts.txt", _("yhosts"));
-		o.value("https://cdn.jsdelivr.net/gh/kongfl888/ad-rules/malhosts.txt", _("Anti-Mal Hosts"));
-		o.value("https://cdn.jsdelivr.net/gh/kongfl888/ad-rules/antihosts-gambling-porn.txt", _("Anti-Gambling-Porn"));
+		o.value("https://fastly.jsdelivr.net/gh/privacy-protection-tools/anti-AD@master/adblock-for-dnsmasq.conf", _("anti-AD (Privacy-Protect|Preferred)"));
+		o.value("https://fastly.jsdelivr.net/gh/AdguardTeam/AdGuardSDNSFilter@gh-pages/Filters/filter.txt", _("AdGuard Filter"));
+		o.value("https://fastly.jsdelivr.net/gh/Cats-Team/AdRules/hosts.txt", _("Cats-Team Hosts"));
+		o.value("https://fastly.jsdelivr.net/gh/VeleSila/yhosts/hosts.txt", _("yhosts"));
+		o.value("https://fastly.jsdelivr.net/gh/kongfl888/ad-rules/malhosts.txt", _("Anti-Mal Hosts"));
+		o.value("https://fastly.jsdelivr.net/gh/kongfl888/ad-rules/antihosts-gambling-porn.txt", _("Anti-Gambling-Porn"));
 		o.value("https://easylist-downloads.adblockplus.org/easylistchina+easylist.txt", _("Easylistchina+Easylist"));
 
 		o = s.option(form.Button, "bfilter");

--- a/luci-app-my-dnshelper/htdocs/luci-static/resources/view/my-dnshelper/settings.js
+++ b/luci-app-my-dnshelper/htdocs/luci-static/resources/view/my-dnshelper/settings.js
@@ -162,11 +162,6 @@ return view.extend({
 		o.rmempty = false;
 		o.default = 'false';
 
-		o = s.option(form.Flag, "block_google", _("Block Google Website and API"));
-		o.description = _("Leave the network for a period of time");
-		o.rmempty = false;
-		o.default = 'false';
-
 		o = s.option(form.Flag, "my_github", _("Easy to Visit GitHub Website"));
 		o.rmempty = false;
 		o.default = 'false';

--- a/my-dnshelper/files/etc/config/my-dnshelper
+++ b/my-dnshelper/files/etc/config/my-dnshelper
@@ -8,7 +8,7 @@ config my-dnshelper
 	option block_games '0'
 	option block_short '0'
 	option block_google '0'
-	option dns_detect '0'
+	option dns_detect '1'
 	option app_test '1'
 	option dns_cache '600'
 	option dns_check '1'
@@ -20,6 +20,5 @@ config my-dnshelper
 	option dns_log '0'
 	option rev_log '0'
 	option my_github '0'
-	option time_update '12'
-
-
+	option time_update '24'
+	list url 'https://fastly.jsdelivr.net/gh/privacy-protection-tools/anti-AD@master/adblock-for-dnsmasq.conf'
